ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/hooks/usePageState.ts

```diff
@@ -0,0 +1,20 @@
+import useURLParameter, { QueryValue } from 'hooks/useURLParameter';
+
+type PageActionResult<T> = {
+    pageAction: T | undefined;
+    setPageAction: (action: T) => void;
+};
+
+function usePageAction<T>(): PageActionResult<T> {
+    const [pageActionParam, setPageActionParam] = useURLParameter('action', undefined);
+
+    const pageAction = pageActionParam as T;
+
+    function setPageAction(action: T): void {
+        setPageActionParam(action as QueryValue);
```

Comment:
Instead of using a type assertion here, you can enforce that the caller always provides the right type of value by changing the hook signature to
```typescript
type PageActionResult<T extends QueryValue> = {}
function usePageAction<T extends QueryValue>(): PageActionResult<T> {}
```
or
```typescript
T extends string
```
if all we ever want to support are strings.

There is similar code in `useURLStringUnion.ts` for the latter case which does some more validation to ensure that we can _never_ get an invalid value at runtime. The main downsides with that hook are that it doesn't currently support `undefined` like this does, and it only supports strings. If we wanted a page action that was something more complex than strings I'm not sure we could validate it in a general way though (at least not with current tooling).